### PR TITLE
feature/add menu product

### PR DESCRIPTION
**Pull request by xidealo**

### CODE REVIEW

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/gallery/GalleryFragment.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                else -> {
+                    LazyVerticalGrid(
+                        columns = GridCells.Fixed(count = 2),
+                        contentPadding = PaddingValues(all = 16.dp),
+                        horizontalArrangement = Arrangement.Absolute.spacedBy(space = 8.dp),
+                        verticalArrangement = Arrangement.Absolute.spacedBy(space = 8.dp)
+                    ) {
+                        items(
+                            state.photos,
+                            key = { photoLink ->
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/gallery/GalleryFragment.kt:71]: In this case we can use just `items(state.photos) { ... }` without key. It compares elements by default.

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/gallery/GalleryViewState.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.bunbeauty.fooddeliveryadmin.screen.gallery
+
+import androidx.compose.runtime.Immutable
+import com.bunbeauty.presentation.viewmodel.base.BaseViewState
+
+@Immutable
+data class GalleryViewState(
+    val photos: List<String>,
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/gallery/GalleryViewState.kt:8]: `ImmutableList`

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/MenuListFragment.kt [hunk 3]
```diff
@@ -85,7 +87,19 @@ class MenuListFragment : BaseFragment<LayoutComposeBinding>() {
...
             onRefresh = viewModel::refreshData,
-            backActionClick = { findNavController().popBackStack() }
+            backActionClick = { findNavController().popBackStack() },
+            actionButton = {
+                if (menuListViewState.state is MenuListViewState.State.Success) {
+                    FloatingButton(
+                        iconId = R.drawable.ic_plus,
+                        textStringId = R.string.action_menu_list_add,
+                        onClick = {
+                            findNavController().navigateSafe(MenuListFragmentDirections.toCreateMenuProductFragment())
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/MenuListFragment.kt:97]: action + event

File: presentation/src/test/java/com/bunbeauty/presentation/EditMenuProductViewModelTest.kt [hunk 4]
```diff
@@ -389,7 +389,7 @@ class EditMenuProductViewModelTest {
             updateMenuProduct()
             consumeEvents(
                 listOf(
-                    EditMenuProductEvent.ShowUpdateProductError(
+                    com.bunbeauty.presentation.feature.menulist.editmenuproduct.EditMenuProductEvent.ShowUpdateProductError(
```
Comment (MaxAstin) [inline presentation/src/test/java/com/bunbeauty/presentation/EditMenuProductViewModelTest.kt:392]: import

File: data/src/main/java/com/bunbeauty/data/repository/CategoryRepository.kt [hunk 5]
```diff
@@ -13,13 +14,21 @@ class CategoryRepository @Inject constructor(
     private val categoryMapper: CategoryMapper
 ) : CategoryRepo {
 
-    override suspend fun refreshCategories(token: String, companyUuid: String) {
-        when (val result = networkConnector.getCategoriesByCompanyUuid(token, companyUuid)) {
+    override suspend fun fetchCategories(token: String, companyUuid: String): List<Category> {
```
Comment (MaxAstin) [inline data/src/main/java/com/bunbeauty/data/repository/CategoryRepository.kt:17]: Let's add cache,
`getCategoryList` method should check cache and then go to server
`getCategoryListFromRemote` method should go to server

**Review by MaxAstin — CHANGES_REQUESTED** [review]
pls fix comments

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/MenuListFragment.kt (hunk 3 above)
Comment (xidealo) [reply app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/MenuListFragment.kt:97]: this is old screen without actions

**Review by MaxAstin — APPROVED** [review]

**Pull request merged**